nengo_spa/algebras/hrr_algebra.py

```diff
@@ -82,6 +91,28 @@ def bind(self, a, b):
             raise ValueError("Inputs must have same length.")
         return np.fft.irfft(np.fft.rfft(a) * np.fft.rfft(b), n=n)
 
+    def fractional_bind(self, v, exponent):
+        r = np.fft.ifft(np.fft.fft(v) ** exponent)
+        if not np.allclose(r.imag, 0):
+            raise ValueError(
```

Comment:
I wonder whether this should really be a hard error. From the error message it seems to me that it is a case one usually wants to avoid, but nothing that mathematically wouldn't work. However, the check of `r.imag` suggests that we get imaginary components when we only want real-valued components.

Reply:
The error message here is confusing, as a complex angle of +/- pi isn't what causes imaginary components. Those are caused by a negative zero or Nyquist frequency. It seems like there are two types of 'degenerate' unitary vectors, ones with ambiguous output (only useful for integer powers) and those that produce output with imaginary components for non-integer powers. For the latter they could have a use when complex valued semantic pointers are implemented, but are not desired when working with purely reals.